src/ng/filter/filters.js

```diff
@@ -331,11 +331,15 @@ function formatNumber(number, pattern, groupSep, decimalSep, fractionSize) {
   }
 }
 
-function padNumber(num, digits, trim) {
+function padNumber(num, digits, trim, negWrap) {
   var neg = '';
   if (num < 0) {
-    neg =  '-';
-    num = -num;
+    if (negWrap) {
+      num = -num + 1;
```

Comment:
Year 0 is 1 BC, year -1 is 2 BC

On Saturday, February 13, 2016, Georgios Kalpakas <notifications@github.com>
wrote:

> In src/ng/filter/filters.js
> <https://github.com/angular/angular.js/pull/14026#discussion_r52828854>:
>
> >    var neg = '';
> >    if (num < 0) {
> > -    neg =  '-';
> > -    num = -num;
> > +    if (negWrap) {
> > +      num = -num + 1;
>
> Is this +1 necessary ? I think it's not consistent with how browsers
> treat negative years.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular.js/pull/14026/files#r52828854>.
>